main.js

```diff
@@ -0,0 +1,120 @@
+const posts = []
+const postList = document.getElementById('post-list');
+const postBtn = document.getElementById('post-btn');
+const lineBreak = document.createElement('br');
+
+function makePost() {
+  const postInput = document.getElementById('post-input');
+  const posterName = document.getElementById('name-input');
+  const postId = Math.random().toString(36).substr(2, 9); 
+  if (postInput.value === "" || posterName.value === '')  {
+    alert('Both input fields must be entered to make a post.')
+    return;
+  } else {
+    const postDiv = document.createElement('div');
+    postDiv.className = 'list-group-item list-group-flush';
+    postDiv.id = `post-${postId}`;
+    postDiv.innerHTML = `
+      <h5><button class='btn btn-info btn-md remove' id='${postId}'>Remove</button> <button class='btn btn-info btn-md comment' id='${postId}'>View Comments</button> ${postInput.value} - Posted By: ${posterName.value}</h5>
+      <div id="comment-section-${postId}" class="${postId} d-none"></div>
+      <input id="comment-text-${postId}" class="form-control comm-text-input" placeholder="Comment Text" />
+      <br>
+      <input id="comment-name-${postId}" class="form-control comm-name-input" placeholder="Your Name" />
+      <br>
+      <button id='${postId}' class="btn btn-primary post-comm">Post Comment</button>
+    `;
+    postList.appendChild(postDiv);
+    posts.push({
+      id: postId,
+      postText: postInput.value,
+      comments: [],
+    });
+    postInput.value = '';
+    posterName.value = '';
+  };
+};
+
+function commentsRenderToggle(postId) {
+  const commentSection = document.querySelector(`#comment-section-${postId}`);
+  if (commentSection.classList.contains('d-none')) {
+    commentSection.classList.remove('d-none');
+  } else {
+    commentSection.classList.add('d-none');
+  };
+};
+
+function postComment(postId) {
+  const commId = Math.random().toString(36).substr(2, 9); // Assigns Specific ID to Comments //
+  const commTextInput = document.querySelector(`#comment-text-${postId}`);
+  const commNameInput = document.querySelector(`#comment-name-${postId}`);
+  const commDiv = document.createElement('div');
+  commDiv.id = commId;
+  commDiv.className = `comment-div ${postId}`;
+  const postIndex = posts.findIndex((post) => post.id === postId);
+  const commentSection = document.querySelector(`#comment-section-${postId}`);
+  // Checks Input Fields //
+  if (commTextInput.value === '' || commNameInput.value === '') {
+    alert('Both input fields must be entered to post a comment.');
+    return;
+  } else {
+    const commData = `
+    <h6><button type='button' class='btn btn-info btn-sm remove-comm' id='${commId}'>Remove</button>
+    ${commTextInput.value} - Posted By: ${commNameInput.value}</h6>`;
+    commentSection.innerHTML = ''
+    commentSection.appendChild(commDiv);
```

Comment:
on each comment you are recreating the whole comments div. 
Better to append just the comment itself.
```
const newComment = document.createElement('h6');
newComment.innerHTML = commData;
commentSection.appendChild(newComment);
```